modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -2268,6 +2269,83 @@ export class Wallet implements IWallet {
     }
   }
 
+  public async buildTokenEnablements(
+    params: BuildTokenEnablementOptions = { tokens: [] }
+  ): Promise<PrebuildTransactionResult[]> {
+    if (!this.baseCoin.requiresTokenEnablement()) {
+      throw new Error(`${this.baseCoin.getFullName()} does not require token enablements`);
+    }
+    if (params.tokens.length === 0) {
+      throw new Error('No tokens are being specified');
+    }
+    if (params.recipients) {
+      throw new Error('Can not specify recipients for token enablement transactions');
+    }
+
+    if (params.reqId) {
+      this.bitgo.setRequestTracer(params.reqId);
+    }
+
+    const buildParams = _.pick(params, this.prebuildWhitelistedParams());
+    buildParams.type = 'enabletoken';
+    // Check if we build with intent
+    if (this._wallet.multisigType === 'tss') {
+      throw new Error('tss not supported for token enablement');
+    } else {
+      // Rewrite tokens into recipients for buildTransaction
+      buildParams.recipients = params.tokens.map((token) => {
+        return { tokenName: token };
+      });
+      return [await this.prebuildTransaction(buildParams)];
+    }
+  }
+
+  public async sendTokenEnablement(params: PrebuildAndSignTransactionOptions = {}): Promise<any> {
+    if (!this.baseCoin.requiresTokenEnablement()) {
+      throw new Error(`${this.baseCoin.getFullName()} does not require token enablement transactions`);
+    }
+
+    if (typeof params.prebuildTx === 'string' || params.prebuildTx === undefined) {
+      throw new Error('Invalid build of token enablement.');
+    }
```

Comment:
is there a way where we can check that the `prebuildTx` is a token enablement txn?

Reply:
`prebuildTransaction` does not return the `buildParams` used so this should be pretty difficult

Reply:
Another thing to note is that this function does not do anything specific to token enablements. It's the same as `sendAccountConsolidation` only we omit a few checks they do

Reply:
👍 thanks for explaining